### PR TITLE
TEZ-4266: Make Tez Inherit From Apache POM

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

File: tez-dist/src/main/assembly/tez-dist-minimal.xml [hunk 1]
```diff
@@ -26,6 +26,7 @@
         <exclude>org.apache.tez:tez-aux-services</exclude>
         <exclude>org.apache.tez:tez-ext-service-tests</exclude>
         <exclude>org.apache.tez:tez-ui</exclude>
+        <exclude>org.apache.tez:tez-docs</exclude>
```
Comment (belugabehr) [inline tez-dist/src/main/assembly/tez-dist-minimal.xml:29]: For some reason, the latest Maven Assembly plugin dies (NPE) when trying to process this module.  I don't think it's required in the minimal distribution anyway.

**Comment by belugabehr** [comment]
@jteagles 

One single unit test failing in the `tez-ui` module.  Any idea what may be causing this?  Hard to see how these changes would affect this.

```
not ok 750 PhantomJS 2.1 - Unit | Serializer | dag: containerLogs test
    ---
        actual: >
            null
        message: >
            Died on test #2 testWrapper@http://localhost:7357/assets/test-support.js:6240:16
            test@http://localhost:7357/assets/test-support.js:6254:44
            http://localhost:7357/assets/tests.js:19223:24
            http://localhost:7357/assets/vendor.js:477:34
            tryFinally@http://localhost:7357/assets/vendor.js:358:21
            requireModule@http://localhost:7357/assets/vendor.js:475:15
            require@http://localhost:7357/assets/test-loader.js:60:16
            loadModules@http://localhost:7357/assets/test-loader.js:51:25
            load@http://localhost:7357/assets/test-loader.js:82:35
            http://localhost:7357/assets/test-support.js:6037:20: undefined is not an object (evaluating 'this.get')
        Log: |
```

**Comment by jteagles** [comment]
Unit test failure in tez-ui is covered by https://issues.apache.org/jira/browse/TEZ-4241 so is unrelated as you have stated. It will take some time to compare before and after this patch. The version of plugins before and after will need to be checked and differences evaluated.

**Comment by belugabehr** [comment]
@jteagles What do you require of me to move this forward?

Thanks for your shepherding.

**Comment by jteagles** [comment]
@belugabehr, the effective-pom and dependency:tree before and after this patch will need to be analyzed. If you are able to provide this analysis, I would greatly appreciate it. If not, I will take a look later this week.

**Comment by belugabehr** [comment]
Visual inspection yields:

```
maven-compiler-plugin 3.1 -> 3.8.1
maven-enforcer-plugin 1.4 -> 1.4.1
maven-jar-plugin 2.4 -> 3.2.0
maven-source-plugin 2.2.1 -> 3.2.1
maven-javadoc-plugin 2.10.4 -> 3.0.1
maven-assembly-plugin 2.4 -> 3.2.0
maven-war-plugin 3.2.3 -> 3.2.2 (regression)
maven-shade-plugin 2.4.3 -> 3.1.1
apache-rat-plugin 0.10 -> 0.13
maven-surefire-plugin 3.0.0-M4-> 3.0.0-M4 (override; higher than Apache POM)
maven-site-plugin 3.4 -> 3.7.1
maven-deploy-plugin 2.8.1 -> 2.8.2
maven-gpg-plugin 1.4 -> 1.6
maven-project-info-reports-plugin 2.9 -> 3.0.0
maven-clean-plugin 3.0.0 -> 3.1.0
maven-enforcer-plugin 1.4.0 -> 1.4.1
```

**Pull request merged**